src/main/java/io/specto/hoverfly/junit/dsl/SingleQuoteHttpBodyConverter.java

```diff
@@ -0,0 +1,60 @@
+package io.specto.hoverfly.junit.dsl;
+
+public class SingleQuoteHttpBodyConverter implements HttpBodyConverter {
+
+    private String convertedBody;
+
+    private SingleQuoteHttpBodyConverter(String json) {
+        this.convertedBody = convert(json);
+    }
+
+    public static SingleQuoteHttpBodyConverter jsonWithSingleQuotes(String body) {
+        return new SingleQuoteHttpBodyConverter(body);
+    }
+
+    @Override
+    public String body() {
+        return convertedBody;
+    }
+
+    @Override
+    public String contentType() {
+        return APPLICATION_JSON;
+    }
+
+    /**
+     * Reads the input text with possible single quotes as delimiters
+     * and returns a String correctly formatted.
+     * <p>For convenience, single quotes as well as double quotes
+     * are allowed to delimit strings. If single quotes are
+     * used, any quotes, single or double, in the string must be
+     * escaped (prepend with a '\').
+     *
+     * @param text the input data
+     * @return String without single quotes
+     */
+    private static String convert(String text) {
```

Comment:
One easy way I can think of it is using a stack to keep track with the single quote rather than the boolean, and check if anything left behind in the stack at the end.